### PR TITLE
guard doc fixes

### PR DESCRIPTION
**Pull request by davidhadas**
Fixes and additions to `guard` documentation related to set and unset TLS
Include also how to install `guard` with the operator
Include also how to install `guard` that uses TLS with the operator

### CODE REVIEW

**Comment by psschwei** [comment]
waiting on https://github.com/knative-sandbox/security-guard/pull/125 to merge

**Comment by davidhadas** [comment]
/cc @houshengbo 
Please review the text and script about operator install with security-guard both with and without TLS

File: docs/serving/app-security/security-guard-install.md [hunk 1]
```diff
@@ -61,15 +61,153 @@ To start this tutorial, after installing Knative Serving, run the following proc
...
+    metadata:
+      name: knative-serving
+      namespace: knative-serving
+    spec:
+      security:
+        securityGuard:
+          enabled: true
+      ingress:
+        kourier:
+          enabled: true
```
Comment (houshengbo) [inline docs/serving/app-security/security-guard-install.md:86]: This is the configuration to use kourier as the ingress. We need to indicate that.
It is possible that users use istio or contour as the ingress, then the configuration is different.

File: docs/serving/app-security/security-guard-install.md [hunk 2]
```diff
@@ -61,15 +61,153 @@ To start this tutorial, after installing Knative Serving, run the following proc
...
+      config:
+        network:
+          ingress.class: "kourier.ingress.networking.knative.dev"
```
Comment (houshengbo) [inline docs/serving/app-security/security-guard-install.md:89]: This is the configuration for the ingress as well.

File: docs/serving/app-security/security-guard-install.md [hunk 3]
```diff
@@ -61,15 +61,153 @@ To start this tutorial, after installing Knative Serving, run the following proc
...
+            value: "true"
+    security:
+        securityGuard:
+        enabled: true
+    ingress:
+        kourier:
+        enabled: true
+    config:
+        network:
+        ingress.class: "kourier.ingress.networking.knative.dev"
```
Comment (houshengbo) [inline docs/serving/app-security/security-guard-install.md:208]: Line 203 to Line 208 are configuration for ingress, do we need to add them for the security guard?

**Comment by davidhadas** [comment]
Two options:
Either we document just the lines that need to change in the operator, or 
We give one example (Kourier) and highlight that for istio and contour changes are needed

I now committed the changes for the second option...

**Comment by houshengbo** [comment]
OK. It sounds good to me.

**Comment by davidhadas** [comment]
> waiting on [knative-sandbox/security-guard#125](https://github.com/knative-sandbox/security-guard/pull/125) to merge

merged

**Comment by davidhadas** [comment]
/hold
(release will change from 0.3 to 0.4)

**Review by psschwei — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**